ivadomed/loader/generalized_loader_configuration.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import copy
+import typing
+from typing import List, Tuple
+
+from ivadomed.keywords import ROIParamsKW
+from ivadomed.transforms import Compose, prepare_transforms
+
+if typing.TYPE_CHECKING:
+    from ivadomed.loader.patch_filter import PatchFilter
+    from ivadomed.loader.slice_filter import SliceFilter
+
+
+class GeneralizedLoaderConfiguration:
```

Comment:
I second @mariehbourget's comment about the naming convention... I'm still not convinced with the name but let's move ahead meanwhile, we could think of a better name. Renaming to be adopted for the class name as well...